classes/item.rb

```diff
@@ -0,0 +1,39 @@
+require 'date'
+
+class Item
+  attr_accessor :archived, :source, :publish_date, :id, :genre, :can_be_archived, :label
+
+  def initialize(publish_date, id = nil)
+    @id = id || Random.rand(1..1000)
+    @publish_date = publish_date
+  end
+
+  def add_genre(genre)
+    @genre = genre
+  end
+
+  def add_author(author)
+    @author = author
+    author.items << self unless author.items.include?(self)
+  end
+
+  def add_label(label)
+    @label = label
+    @label.add_item(self)
+  end
```

Comment:
- After fixing the custom setter method correctly, please implement the belongs-to relationship between the label class and the book class as you did for the author class and source class. i.e add the line below inside the custom setter method
```ruby
    label.items << self unless label.items.include?(self)
```